web-marketplace/src/components/organisms/LoginButton/hooks/useSocialProviders.ts

```diff
@@ -2,7 +2,10 @@ import { apiUri } from 'lib/apiUri';
 import { GoogleLoginEvent } from 'lib/tracker/types';
 import { useTracker } from 'lib/tracker/useTracker';
 
-export const useSocialProviders = (createProject: boolean = false) => {
+export const useSocialProviders = (
+  createProject: boolean = false,
+  buyCreditsFrom?: string,
```

Comment:
since we're adding more custom redirect routes, maybe we should just refactor this to take an optional `route` as param?

Reply:
Sounds good to me, passing `redirectRoute` as a param definitely makes sense!

Reply:
This [comment](https://github.com/regen-network/regen-web/pull/2494#discussion_r1786250222) now should be: 
```
<LoginFlow
  isModalOpen={isModalOpen}
  onModalClose={onModalClose}
  wallets={walletsUiConfig}
  modalState={modalState}
  # New prop:
  redirectRoute={`${projectHref}/buy`}
/>
```